// react-native-sf-ums/src/SFUM.js
/**
 * Created by SF on 2018/4/8.
 */

import React, {Component} from 'react';
import {
    NativeModules,
} from 'react-native';
var ShareUtil = NativeModules.UMShareModule;
var PushUtil = NativeModules.UMPushModule;
var AnalyticsUtil = NativeModules.UMAnalyticsModule;

export default class SFUM extends Component{

    /*** TODO: 分享平台代码 (platfrom)
     *  0	QQ
     *  1	SINA
     *  2	微信
     *  3	朋友圈
     *  4	QQ空间
     **/


    /*** TODO: 登陆授权
     *  1. platform  授权平台  见上方
     *  2. callback 成功回调
     **/
    static login(plat, callback) {
        ShareUtil.auth(plat, (code, result, message) => {
            console.log(code,message);
            callback(result);
        });
    }

    /*** TODO: 只有分享功能 没有界面
     *  1. 只传文字,为分享文字 (只填txt)
     *  2. 只传图片, 既分享图片 (只填img)
     *  3. 分享链接需要填url (其他形式不用填)
     *  4. 分享url 可填title 或者 不填 (txt字段无效)
     *  5. platform  分享的平台  见上方
     *  6. callback 成功回调
     * */

    static share(text, img, url, title, platform, callback) {
        ShareUtil.share(text, img, url, title, platform, (code, message) => {
            callback(code, message);
        });
    }


    /*** TODO: 带界面的分享
     *  1. 只传文字,为分享文字 (只填txt)
     *  2. 只传图片, 既分享图片 (只填img)
     *  3. 分享链接需要填url (其他形式不用填)
     *  4. 分享url 可填title 或者 不填 (txt字段无效)
     *  5. platArray  分享的平台 默认是QQ 微信 微博 , 见上方
     * */

    static shareBoard(txt, img, url, title, platArr) {
        var list = [0, 1, 2, 3, 4];
        var txt = txt;
        var img = img;
        var url = url;
        var title = title;
        ShareUtil.shareboard(txt, img, url, title, platArr ? platArr : list, (code, message) => {
            console.log(code, message);
        });
    }

    /*** TODO: 添加标签
     *  推送的tag
     * */
    static addTag(tag) {
        PushUtil.addTag(tag,(code,remain) =>{
            console.log(code,remain);
        })
    }

    /*** TODO: 删除标签
     *  删除的tag
     * */
    static delTag(tag) {
        PushUtil.deleteTag(tag,(code,result) =>{
            console.log(tag,result);
        })
    }

    /*** TODO: 显示所有标签
     *  显示所有的tag
     * */
    static showTag(callback) {
        PushUtil.listTag((code,result) =>{
            callback(result);
        })
    }

    /*** TODO: 添加别名
     *  添加别名
     *  alias: ID
     *  type: 别名类型
     * */
    static addAlias(alias,type) {
        PushUtil.addAlias(alias,type,(code) =>{
            console.log(code);
        })
    }

    /*** TODO: 删除别名
     *  删除别名
     *  alias: ID
     *  type: 别名类型
     * */
    static deleteAlias(alias,type) {
        PushUtil.deleteAlias(alias,type,(code) =>{

        })
    }


    static appInfo(callback) {
        PushUtil.appInfo((result) =>{
            callback(result);
        })
    }


    //统计 (附加功能,统计开启次数在主工程)
    static eventNum(eventId) {
        AnalyticsUtil.onEvent(eventId);
    }

    /*** TODO: 统计用户ID
     *  eventId: 用户ID
     *  eventLabel: 用户分类
     * */
    static eventNumWithAlias(eventId,eventLabel) {
        AnalyticsUtil.onEventWithLabel(eventId,eventLabel);
    }

    /*** TODO: 统计用户ID
     *  eventId: 用户ID
     *  eventLabel: 用户分类
     * */
    //eventData 为当前事件的属性和取值（键值对），不能为空，如：{name:”umeng”,sex:”man”}
    static eventNumWithDic(eventId,eventData) {
        AnalyticsUtil.onEventWithMap(eventId,eventData);
    }


    static eventWithMap(eventId,eventData,eventNum) {
        AnalyticsUtil.onEventWithMapAndCount(eventId,eventData,eventNum);
    }


    static profileSignInWithPUID(puid) {
        AnalyticsUtil.profileSignInWithPUID(puid);
    }

    static profileSignOff(){
        AnalyticsUtil.profileSignOff()
    }



}
